### PR TITLE
Fixes for editShow

### LINKED ISSUE #4072: Loading editShow whows wrong location
Reminder for @OmgImAlexis 

When editing a show, it's using the location from the browsers localStorage.
Sometime it's using the wrong location, sometime (when starting clean) it's showing an empty input box.

Note!
I renamed the 'key' attribute of the file-browser component props, int `localStorageKey` because vue was complaining, as it's a reserved keyword. (that's also now in develop).

I'm not sure if that's the cause. I don't think so.

### LINKED ISSUE #4076: Propers - AttributeError: 'unicode' object has no attribute 'series_name'
### Before submitting your issue:

Enable debug logging in Medusa settings, reproduce the error (be sure to disable after the bug is fixed)
**Branch/Commit:** develop
**OS:** Win7
**Logs:**
```
2018-04-22 19:13:13 DEBUG    FINDPROPERS :: [6d54662] Starting new thread: FINDPROPERS
2018-04-22 19:13:13 INFO     FINDPROPERS :: [6d54662] Beginning the search for new propers
2018-04-22 19:13:13 INFO     FINDPROPERS :: [6d54662] Using proper search days: 2
2018-04-22 19:13:13 INFO     FINDPROPERS :: [Rarbg] :: [6d54662] Searching for any new PROPER releases from Rarbg
2018-04-22 19:13:13 DEBUG    FINDPROPERS :: [Rarbg] :: [6d54662] 295647: Setter sets location to E:\TV\Blindspot\Season 3\Blindspot.S03E18.PROPER.720p.HDTV.x264-KILLERS.mkv
2018-04-22 19:13:13 DEBUG    FINDPROPERS :: [Rarbg] :: [6d54662] Checking if E:\TV\Blindspot\Season 3\Blindspot.S03E18.PROPER.720p.HDTV.x264-KILLERS.nfo exists: True
2018-04-22 19:13:13 DEBUG    FINDPROPERS :: [Rarbg] :: [6d54662] Checking if E:\TV\Blindspot\Season 3\Blindspot.S03E18.PROPER.720p.HDTV.x264-KILLERS-thumb.jpg exists: True
2018-04-22 19:13:14 DEBUG    FINDPROPERS :: [Rarbg] :: [6d54662] GET URL: http://torrentapi.org/pubapi_v2.php?get_token=get_token&app_id=medusa&format=json [Status: 200]
2018-04-22 19:13:14 DEBUG    FINDPROPERS :: [Rarbg] :: [6d54662] User-Agent: Medusa/0.2.2 (Windows; 7; 83ab66f0-4626-11e8-99e3-001fd0d1e3a5)
2018-04-22 19:13:14 DEBUG    FINDPROPERS :: [Rarbg] :: [6d54662] Search mode: Episode
2018-04-22 19:13:14 DEBUG    FINDPROPERS :: [Rarbg] :: [6d54662] Search string: Blindspot S03E18 {{PROPER|REPACK|REAL|RERIP}}
2018-04-22 19:13:14 DEBUG    FINDPROPERS :: [Rarbg] :: [6d54662] Searching only ranked torrents
2018-04-22 19:13:19 DEBUG    FINDPROPERS :: [Rarbg] :: [6d54662] GET URL: http://torrentapi.org/pubapi_v2.php?sort=seeders&min_leechers=0&min_seeders=1&search_tvdb=295647&app_id=medusa&search_string=Blindspot+S03E18+%7B%7BPROPER%7CREPACK%7CREAL%7CRERIP%7D%7D&ranked=1&category=tv&format=json_extended&token=l7huma64or&limit=100&mode=search [Status: 200]
2018-04-22 19:13:19 DEBUG    FINDPROPERS :: [Rarbg] :: [6d54662] User-Agent: Medusa/0.2.2 (Windows; 7; 83ab66f0-4626-11e8-99e3-001fd0d1e3a5)
2018-04-22 19:13:19 DEBUG    FINDPROPERS :: [Rarbg] :: [6d54662] Found result: Blindspot.S03E18.PROPER.HDTV.x264-KILLERS[rartv] with 1049 seeders and 271 leechers
2018-04-22 19:13:19 DEBUG    FINDPROPERS :: [Rarbg] :: [6d54662] Found result: Blindspot.S03E18.PROPER.720p.HDTV.x264-KILLERS[rartv] with 695 seeders and 99 leechers
2018-04-22 19:13:19 ERROR    FINDPROPERS :: [Rarbg] :: [6d54662] Exception generated: 'unicode' object has no attribute 'series_name'
Traceback (most recent call last):
  File "C:\Medusa\medusa\scheduler.py", line 93, in run
    self.action.run(self.force)
  File "C:\Medusa\medusa\search\proper.py", line 67, in run
    propers = self._get_proper_results()
  File "C:\Medusa\medusa\search\proper.py", line 126, in _get_proper_results
    cur_propers = cur_provider.find_propers(recently_aired)
  File "C:\Medusa\medusa\providers\generic_provider.py", line 203, in find_propers
    search_strings = self._get_episode_search_strings(episode_obj, add_string=term)
  File "C:\Medusa\medusa\providers\generic_provider.py", line 621, in _get_episode_search_strings
    all_possible_show_names = episode.series.get_all_possible_names()
  File "C:\Medusa\medusa\tv\series.py", line 2139, in get_all_possible_names
    show_names = {exception.series_name for exception in get_scene_exceptions(self, season)}
  File "C:\Medusa\medusa\tv\series.py", line 2139, in <setcomp>
    show_names = {exception.series_name for exception in get_scene_exceptions(self, season)}
AttributeError: 'unicode' object has no attribute 'series_name'
```

### PR DESCRIPTION
**Pull request by sharkykh**
To do:
- [x] Fix show location file browser, autocomplete, saving location.
       Not every file browser needs a localStorage key. Show location doesn't (removed)
       Closes #4072 
- [x] Sync changes to slim + dark themes
- [x] Fix #4076 
- [x] API has a patch for `series.allowed_qualities` and `series.preferred_qualities`, yet those properties have no `@[property].setter` function in `medusa/tv/series.py@Series`
- [x] Fix saving black/white lists.
- [x] Fix saving default episode status - currently not being saved at all.
- [x] Restore POST functionally to editShow because [**massEditSubmit still uses it**](https://github.com/pymedusa/Medusa/blob/130cfca37762342537d7d19720e42635a2fbe7b9/medusa/server/web/manage/handler.py#L681-L688) ~(all themes), and 
because **legacy theme** still uses that method for saving shows.~

### CODE REVIEW

File: themes/light/templates/vue-components/file-browser.mako [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <script type="text/x-template" id="file-browser-template">
     <div>
-        <input v-model="currentPath" :name="name" type="text" id="episodeDir" class="input-sm form-control form-control-inline fileBrowserField"/>
-        <input v-if="showBrowseButton" type="button" :value="'Browse\u2026'" class="btn btn-inline fileBrowserButton"/>
-        <div class="fileBrowserDialog" style="display:hidden"></div>
+        <input v-model="currentPath" ref="locationInput" :name="name" @keyup.enter="browse($event.target.value)" type="text" class="input-sm form-control fileBrowserField" style="width: 70%;" />
```
Comment (sharkykh) [inline themes/light/templates/vue-components/file-browser.mako]: This `@keyup.enter="browse($event.target.value)"` might have made it in by mistake, I'll have to test.

File: themes/light/templates/vue-components/file-browser.mako [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+        <input v-if="showBrowseButton" type="button" :value="'Browse\u2026'" class="btn fileBrowserButton"/>
+        <div class="fileBrowserDialog" style="display: none;"></div>
```
Comment (sharkykh) [inline themes/light/templates/vue-components/file-browser.mako]: The style changes fix the `Browse...` button location.

File: themes/light/templates/vue-components/file-browser.mako [hunk 3]
```diff
@@ -262,11 +267,22 @@ Vue.component('file-browser', {
             return resultField;
         };
 
-        $(this.$el).children('input').fileBrowser({
+        $(this.$refs.locationInput).fileBrowser({
```
Comment (sharkykh) [inline themes/light/templates/vue-components/file-browser.mako]: This was linking the file browser to 3 elements including the browse button, causing some issues with the autocomplete, if I remember the issue correctly.

File: themes/light/templates/editShow.mako [hunk 4]
```diff
@@ -157,15 +149,11 @@ const startVue = () => {
...
-            saveLocation: function(value) {
-                this.series.config.location = value;
-            },
             updateLanguage: function(value) {
                 this.series.language = value;
             },
             combineQualities() {
-                const reducer = (accumulator, currentValue) => accumulator + currentValue;
-                
+                const reducer = (accumulator, currentValue) => accumulator | currentValue;
```
[the remark this answers is not in the document]
Comment (sharkykh) [reply themes/light/templates/editShow.mako]: @medariox

[the remark this answers is not in the document]
Comment (medariox) [reply themes/light/templates/editShow.mako]: Yeah, I agree with @sharkykh - if we want to have it exactly as the python code, this change is needed.

**Comment by sharkykh** [comment]
@p0psicles  Sorry for changing a lot. I will help with the rebase of your PR if want me to.
Unfortunately I'm still not completely happy with what I managed to do, but I'd rather this gets merged for now, as it looks stable to me.

**Comment by p0psicles** [comment]
No prob. I created mine with the expectation that I needed to rebase anyway. No need to help. Lets merge this if your done for now

File: themes-default/slim/views/editShow.mako [hunk 5]
```diff
@@ -133,40 +113,40 @@ const startVue = () => {
                     }
                 };
             },
-            onChangeIgnoredWords: function(items) {
+            onChangeIgnoredWords(items) {
 		        console.debug('Event from child component emitted', items);
```
Comment (OmgImAlexis) [inline themes-default/slim/views/editShow.mako]: You can remove all the `console.debug` statements.

@p0psicles if you need to debug events use the Vue Devtools Chrome extension.

**Comment by fernandog** [comment]
@sharkykh thanks the `Series.get_all_possible_names: Fix AttributeError` fixed my FINDPROPERS crashing

**Pull request merged**